src/components/ContactPreview.tsx

```diff
@@ -0,0 +1,50 @@
+'use client'
+
+import { css } from '../../styled-system/css'
+import useAvatar from '@/components/useAvatar'
+import { useRouter } from 'next/navigation'
+import { createClientComponentClient } from '@supabase/auth-helpers-nextjs'
+import { Database } from '@/types/supabase'
+import { useContext } from 'react'
+import { UserContext } from '@/components/UserProvider'
+import { Profile } from '@/types/models'
+
+type Props = {
+  contact: Profile
+}
+
+export default function ContactPreview({ contact }: Props) {
+  const supabase = createClientComponentClient<Database>()
+  const router = useRouter()
+  const { user } = useContext(UserContext)
+
+  const { AvatarPreview } = useAvatar({
+    avatarUrl: contact.avatar_url,
+  })
+
+  const handleClick = async () => {
+    // TODO: check the chatId not to create multiple chats for the same users
+    const { data, error } = await supabase
+      .from('chats')
+      .insert({
+        users: [user?.id!, contact.id],
+      })
+      .select()
+      .single()
+    if (error) {
+      return
+    }
+    router.push(`/messages/${data.chat_id}`)
+  }
+
+  return (
+    // onClick on a div 🤔
```

Comment:
ai ai ai... I taught you better than this! Why a div? In your TODO above you could check that on mount and if it already exists it's a simple <a>, otherwise is a button. You could even split the UI into 2 lists:
- First the people you have a chat with. 
- Then all the remaining profiles.

Reply:
hehe true, I think I forgot the TODO, updated